pkg/webhook/csidriver/csidriver.go

```diff
@@ -47,12 +49,29 @@ func (s *SharedResourcesCSIDriverWebhook) Name() string { return WebhookName }
 
 // Validate if the incoming request even valid
 func (s *SharedResourcesCSIDriverWebhook) Validate(req admissionctl.Request) bool {
-	return req.Kind.Kind == "Pod"
+	switch req.Kind.Kind {
+	case "Pod":
+		return req.Kind.Kind == "Pod"
+	case "SharedConfigMap":
+		return req.Kind.Kind == "SharedConfigMap"
+	case "SharedSecret":
+		return req.Kind.Kind == "SharedSecret"
+	}
+	return false
 }
 
 // Authorized implements Webhook interface
 func (s *SharedResourcesCSIDriverWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
-	return s.authorizePod(request)
+	var ret admissionctl.Response
+	switch request.Kind.Kind {
```

Comment:
You should probably add a default case here which returns a Denied response and logs some kind of warning about a Kind that is not handled